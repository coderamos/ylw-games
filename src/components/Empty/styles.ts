import styled, { css } from 'styled-components';

import { ColorStyleProps } from 'types/global';

type ColorProps = {
  color: ColorStyleProps;
};

export const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
`;

export const Image = styled.img`
  max-width: 100%;
`;

export const Title = styled.h2<ColorProps>`
  ${({ theme, color }) => css`
    color: ${color !== 'primary' ? theme.colors.secondary : theme.colors.white};
    font-size: ${theme.font.sizes.xxlarge};
  `}
`;

export const Description = styled.p<ColorProps>`
  ${({ theme, color }) => css`
    color: ${color === 'primary' ? theme.colors.white : theme.colors.secondary};
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.light};
    margin-bottom: ${theme.spacings.medium};
  `}
`;
