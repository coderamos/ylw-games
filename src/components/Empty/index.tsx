import Button from 'components/Button';
import Link from 'next/link';

import { ColorStyleProps } from 'types/global';

import * as s from './styles';

export type EmptyProps = {
  title: string;
  description: string;
  hasLink?: boolean;
  contentStyle?: ColorStyleProps;
};

const Empty = ({
  title,
  description,
  hasLink,
  contentStyle = 'primary'
}: EmptyProps) => (
  <s.Wrapper>
    <s.Image src="/img/empty.png" alt="hanging joysticks" role="image" />
    <s.Title>{title}</s.Title>
    <s.Description>{description}</s.Description>
    {hasLink && (
      <Link href="/" passHref>
        <Button as="a" colorStyle={contentStyle}>
          go back to store
        </Button>
      </Link>
    )}
  </s.Wrapper>
);

export default Empty;
