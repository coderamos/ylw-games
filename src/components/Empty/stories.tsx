import { Story, Meta } from '@storybook/react/types-6-0';

import Empty, { EmptyProps } from '.';

export default {
  title: 'Design System/Empty',
  component: Empty,
  args: {
    title: 'your wishlist is empty!',
    description: 'games added to your wishlist will apperar here',
    hasLink: true
  },
  parameters: {
    backgrounds: {
      default: 'dark'
    },
    jest: ['Empty.test.tsx']
  }
} as Meta;

export const EmptyDefault: Story<EmptyProps> = args => <Empty {...args} />;
