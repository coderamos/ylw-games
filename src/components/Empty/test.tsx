import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Empty from '.';

const emptyProps = {
  title: 'a example title',
  description: 'a example description'
};

describe('<Empty />', () => {
  it('should render empty component correctly', () => {
    const { container } = renderWithTheme(<Empty {...emptyProps} hasLink />);

    expect(
      screen.getByRole('image', { name: /hanging joysticks/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: /a example title/i })
    ).toBeInTheDocument();
    expect(screen.getByText(/a example description/i)).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /go back to store/i })
    ).toHaveAttribute('href', '/');

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not should render link when hasLink is false', () => {
    renderWithTheme(<Empty {...emptyProps} />);

    expect(
      screen.queryByRole('link', { name: /go back to store/i })
    ).not.toBeInTheDocument();
  });
});
