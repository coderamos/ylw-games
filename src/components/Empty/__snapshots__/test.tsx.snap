// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render empty component correctly 1`] = `
.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.4rem;
  border: none;
  cursor: pointer;
  padding: 0.8rem;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.4rem;
  height: 4rem;
  padding: 0.8rem 3.2rem;
  background: #FFCB00;
  color: #171717;
}

.c4:hover {
  background: #ffd533;
  color: #171717;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 100%;
}

.c2 {
  color: #FFFFFF;
  font-size: 2.8rem;
}

.c3 {
  color: #FFFFFF;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

<main
  class="c0"
>
  <img
    alt="hanging joysticks"
    class="c1"
    role="image"
    src="/img/empty.png"
  />
  <h2
    class="c2"
  >
    a example title
  </h2>
  <p
    class="c3"
  >
    a example description
  </p>
  <a
    class="c4"
    href="/"
  >
    <span
      class=""
    >
      go back to store
    </span>
  </a>
</main>
`;
